leptonai/queue.py

```diff
@@ -0,0 +1,203 @@
+from leptonai.api import workspace as workspace_api
+
+# If not ready, wait for this amount of seconds before checking again.
+_lepton_readiness_wait_time_ = 10
+# Timeout limit.
+_lepton_readiness_timeout_ = 60 * 5  # 5 minutes
+
+# max key and value lengths: 256 bytes for key, and 256KB for value.
+_lepton_max_key_length_ = 256
+_lepton_max_value_length_ = 256 * 1024
+
+
+class Queue(object):
+    """
+    Queue. doc tbd.
+    """
+
+    @staticmethod
+    def create_queue(name: str, conn: Optional[Connection] = None) -> "Queue":
+        """
+        Create a queue.
+        """
+        queue_instance = Queue(
+            name,
+            create_if_not_exists=True,
+            error_if_exists=True,
+            conn=conn,
+        )
+        return queue_instance
+
+    @staticmethod
+    def get_queue(name: str, conn: Optional[Connection] = None) -> "Queue":
+        """
+        Get an existing queue.
+        """
+        kv_instance = Queue(name, create_if_not_exists=False, conn=conn)
+        return kv_instance
+
+    @staticmethod
+    def delete_queue(name: Union["Queue", str], conn: Optional[Connection] = None):
+        """
+        Delete a queue.
+        """
+        conn = conn if conn else workspace_api.current_connection()
+        if isinstance(name, Queue):
+            name = name._queue
+        res = queue_api.delete_queue(conn, name)
+        if not res.ok:
+            raise RuntimeError(
+                f"Failed to delete queue {name}. Error:"
+                f" {res.status_code} {res.content}."
+            )
+
+    @staticmethod
+    def list_queue(conn: Optional[Connection] = None):
+        """
+        List queues in the current workspace.
+        """
+        conn = conn if conn else workspace_api.current_connection()
+        res = queue_api.list_queue(conn)
+        if not res.ok:
+            raise RuntimeError(
+                f"Failed to list queues. Error: {res.status_code} {res.content}."
+            )
+        else:
+            return [s["name"] for s in res.json()]
+
+    def __init__(
+        self,
+        name: str,
+        create_if_not_exists: bool = False,
+        error_if_exists: bool = False,
+        conn: Optional[Connection] = None,
+    ):
+        """
+        Initializes a queue.
+
+        :param str name: the name of the queue
+        :param bool create_if_not_exists: if True, create the queue if it does not exist
+        :param bool error_if_exists: if True, raise an error if the queue already exists
+        :param Connection conn: the connection to use. If None, use the default workspace connection.
+        """
+        self._conn = conn if conn else workspace_api.current_connection()
+        res = queue_api.list_queue(self._conn)
+        if not res.ok:
+            raise RuntimeError(
+                "Failed to access queue server. Error:"
+                f" {res.status_code} {res.content}."
+            )
+        existing_queues = [s["name"] for s in res.json()]
```

Comment:
maybe better to use `existing_queue = Queue.list_queue(self._conn)`

Reply:
will do